src/shared/firewall-util.c

```diff
@@ -48,15 +48,20 @@ FirewallContext *fw_ctx_free(FirewallContext *ctx) {
 }
 
 int fw_add_masquerade(
-                FirewallContext *ctx,
+                FirewallContext **fw_ctx,
                 bool add,
                 int af,
                 const union in_addr_union *source,
                 unsigned source_prefixlen) {
+        FirewallContext *ctx;
 
-        if (!ctx)
-                return -ENOTCONN;
+        if (!*fw_ctx) {
+                int ret = fw_ctx_new(fw_ctx);
```

Comment:
doesn't matter, but we almost always call the variable we collect errors in `int r` and declare it at the outermost function scope. might be something to follow here too, just keep surprise minimal